### ISSUE TITLE
[Clang-Tidy] `// NOLINTNEXTLINE` doesn't work if a method call is split on multiple lines

### ISSUE THREAD

**Issue by H-G-Hristov** [llvm/llvm-project#60779]
This doesn't work but I think it should.

```c++
            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
            CFDictionaryGetKeysAndValues(
                valueOpt->Get(),
                reinterpret_cast<const void**>(keys.data()),
                reinterpret_cast<const void**>(values.data()));
```

**Comment by ADKaster** [comment]
Or perhaps
```
            // NOLINTBEGIN(cppcoreguidelines-pro-type-reinterpret-cast)
            CFDictionaryGetKeysAndValues(
                valueOpt->Get(),
                reinterpret_cast<const void**>(keys.data()),
                reinterpret_cast<const void**>(values.data()));
            // NOLINTEND(cppcoreguidelines-pro-type-reinterpret-cast)
```

**Comment by H-G-Hristov** [comment]
That's a bit excessive just because a single line was wrapped. Maybe a new rule like // NOLINTNEXTEXPRESSION.....

**Comment by PiotrZSL** [comment]
I thing that at the moment where filters are applied, we don't have any info about expression.
And here issue is for reinterpret_cast expr, but next expression is call to CFDictionaryGetKeysAndValues, so nolint next expression could make things more complex things, they would need to be handled on check level, and that would make checks even more slower.

warnings are raised for specific line number, and nolints are maching those lines

**Comment by carlosgalvezp** [comment]
I believe BEGIN/END would be the preferred solution for your use case. All NOLINT expressions are line number-based, without semantic information, so it would be good to keep that consistency and simplicity. Similar behavior exists on other static analyzers and compilers (pragma push/pop).
